x-pack/plugins/spaces/server/lib/create_default_space.ts

```diff
@@ -5,15 +5,23 @@
  */
 
 import { i18n } from '@kbn/i18n';
-import { getClient } from '../../../../server/lib/get_client_shield';
+
+import { first } from 'rxjs/operators';
+import { SavedObjectsService } from 'src/legacy/server/kbn_server';
+import { ElasticsearchServiceSetup } from 'src/core/server';
 import { DEFAULT_SPACE_ID } from '../../common/constants';
 
-export async function createDefaultSpace(server: any) {
-  const { callWithInternalUser: callCluster } = getClient(server);
+interface Deps {
+  elasticsearch: ElasticsearchServiceSetup;
+  savedObjects: SavedObjectsService;
+}
+
+export async function createDefaultSpace({ elasticsearch, savedObjects }: Deps) {
+  const { getSavedObjectsRepository, SavedObjectsClient } = savedObjects;
 
-  const { getSavedObjectsRepository, SavedObjectsClient } = server.savedObjects;
+  const client = await elasticsearch.dataClient$.pipe(first()).toPromise();
```

Comment:
@kobelb this is the usage of `first()` that I had mentioned the other day. If you have a chance to try to tackle this, or give general guidance for how I can, I'd appreciate it!

I stumbled on this because the `defaultSpaceExists` check short-circuits the creation of the default space, but I was under the impression that the consumer of this function (which is still in legacy land) needs something to `await`, so the outcome of this function should still be a promise. If that's not the case, then maybe that simplifies things, but that's where I was getting stuck.

Reply:
> I stumbled on this because the defaultSpaceExists check short-circuits the creation of the default space, but I was under the impression that the consumer of this function (which is still in legacy land) needs something to await, so the outcome of this function should still be a promise. If that's not the case, then maybe that simplifies things, but that's where I was getting stuck.

Since this is an async function, even if we return immediately, it'll still be a promise and everything will behave "alright". The biggest drawback that we get from the current implementation being promise based is that we don't get cancellation. If we were to rewrite `watchStatusAndLicenseToInitialize` to work with observables, we'd be able to cancel it half-way through. Right now, because we break out of the RxJs world, we don't get this...